### PR TITLE
Add checksum validation functionality

### PR DESCRIPTION
**Pull request by maf-embraceit**

### CODE REVIEW

**Review by shahbaz-mehar — COMMENTED** [review]
Changes have been reviewed.

File: admin/controller/altapay/templates/admin/view/altapay.twig [hunk 1]
```diff
@@ -32,9 +32,18 @@
             </div>
           </div>
           <div class="form-group">
-            <label class="col-sm-2 control-label" for="input-title">{{ entry_custom_message }}</label>
+            <label class="col-sm-2 control-label" for="input-secret">{{ entry_secret }}</label>
```
[the remark this answers is not in the document]
Comment (maf-embraceit) [reply admin/controller/altapay/templates/admin/view/altapay.twig:35]: Yes, they can view it in admin panel. I've set it to password type input.

[the remark this answers is not in the document]
Comment (emicha) [reply admin/controller/altapay/templates/admin/view/altapay.twig:35]: @maf-embraceit Would they be able to inspect it and see the actual value?

[the remark this answers is not in the document]
Comment (maf-embraceit) [reply admin/controller/altapay/templates/admin/view/altapay.twig:35]: Yes, @emicha, only if they change the input type using the browser inspect tool.

[the remark this answers is not in the document]
Comment (emicha) [reply admin/controller/altapay/templates/admin/view/altapay.twig:35]: We should avoid that, a secret can't be seen or read, only modified.

Showing the input type as password is not enough, the password needs to be masked, and when they user tries to modify it, the input should be cleared out.

[the remark this answers is not in the document]
Comment (maf-embraceit) [reply admin/controller/altapay/templates/admin/view/altapay.twig:35]: I've resolved the issue in the [commit](https://github.com/AltaPay/plugin-opencart3/pull/32/commits/38743ee7114bea7ce21204fb95e574e0568956c5) by masking the gateway password & terminal secrets if not empty.

File: admin/controller/altapay/templates/admin/controller/altapay.twig.php [hunk 2]
```diff
@@ -181,6 +184,10 @@ public function index()
             $data['payment_altapay_{key}_currency_id'] = $this->config->get('payment_Altapay_{key}_currency_id');
         }
 
+        if (!empty($data['payment_altapay_{key}_secret'])) {
+            $data['payment_altapay_{key}_secret'] = '*****';
+        }
+
```
Comment (emicha) [inline admin/controller/altapay/templates/admin/controller/altapay.twig.php:190]: If the merchant changes another parameter of the account profile, would the password then be reset to `*****`, effectively disabling the user's access, and potentially, letting an attacker login using `*****` ?

**Review by emicha — APPROVED** [review]

**Pull request merged**